src/utils/detect-server.js

```diff
@@ -126,7 +126,7 @@ module.exports.serverSettings = async (devConfig, flags, projectDir, log) => {
   settings.port = port
 
   settings.jwtRolePath = devConfig.jwtRolePath || 'app_metadata.authorization.roles'
-  settings.functionsPort = await getPort({ port: settings.functionsPort || 34567 })
```

Comment:
When running two instance of `netlify dev` around the same time `getPort` things that port `34567` is available which causes one of them to fail.
Falling back to `0` port makes sure that we get a random port for `functionPort` on every instance.